### DISCUSSION TITLE
How to name (multiple) windows inside tabs

### DISCUSSION THREAD

**Discussion by Strykar** [kovidgoyal/kitty#7640] [Q&A] [answered]
![Screenshot from 2024-07-17 21-52-33](https://github.com/user-attachments/assets/fa16b508-a492-4ffd-ac44-9708937945e1)

I run multiple windows inside a tab that are viewing remote logs from different servers.
Is it possible to name each window so I can tell which log window belongs to which server?

**Comment by kovidgoyal** [comment]
Use the set window title action or remote control feature or session
script, whichever fits your workflow better.

**Comment by Strykar** [comment]
I use a session file to setup the tabs and windows and I launch kitty on Gnome login.
Seems `kitty --title Kitty` will do it, but how do I specify this in the session config which looks like:
```
new_tab ⧫🍁⟠
layout vertical
launch bash -c 'ssh host1 /usr/bin/journalctl foo'
launch bash -c 'ssh host2 /usr/bin/journalctl foo'
```
I'm unsure how this works with `launch`.

**Comment by kovidgoyal** [comment]
Don't use title use --class and the sessoins example in the docs shows
you how to set that.

**Comment by Strykar** [comment]
Thanks, I tried the example from the docs but the windows in the tab are unchanged, what am I missing here?
```
# kitty session file
# Set the --class for the new OS window
os_window_class mumbai launch bash..
```

**Reply by kovidgoyal** [reply]
You want

launch --title whatever bash -c ...

See https://sw.kovidgoyal.net/kitty/launch/

**Comment by Strykar** [comment]
The windows have no titles even with launch --title foo bash -c..
https://i.imgur.com/TUaTJoX.png

**Comment by kovidgoyal** [comment, marked as answer]
Sorry, I have no clue what you are trying to do and what you are
actually doing.

Window titles are not displayed anywhere. The tab bar shows the title of
the currently active window. The OS Window titlebar show the title of
the currently active window in the currently active tab.

When you launch using --title the window will have its title permanently
set to whatever you specify. Nothing running in the window can change
it.

kitty --session <(echo -e "launch --title works\nlaunch --title works2")

works as expected.

Good luck.

**Comment by Strykar** [comment]
Ah, for some reason I thought each window could have its name overlaid somewhere within the window itself.
I understand now that the window names can only be shown in the same place as the tab name.

I removed the fixed name I had for that tab and now I can see my custom titles when I mouseover the windows in that tab.
Thanks and sorry about the confusion!
